hyperspy/_signals/signal2d.py

```diff
@@ -95,8 +95,11 @@ def fft_correlation(in1, in2, normalize=False):
     s1 = np.array(in1.shape)
     s2 = np.array(in2.shape)
     size = s1 + s2 - 1
-    # Use 2**n-sized FFT
-    fsize = (2 ** np.ceil(np.log2(size))).astype("int")
+
+    # Calculate optimal FFT size
+    complex_result = (in1.dtype.kind == 'c' or in2.dtype.kind == 'c')
+    fsize = [sp.fft.next_fast_len(a, not complex_result) for a in size]
```

Comment:
May need to replace this with `scipy.fftpack.next_fast_len` for compatibility with scipy<1.4.0